### PR TITLE
Added strings/wordCount

### PR DESCRIPTION
**Pull request by dshubhadeep**

### CODE REVIEW

File: strings/word_count.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+const cleanString = (input: string): string =>
+  input.trim().replace(/\s+/g, " ");
+
+/** Helper to count words in a given string */
+export function wordCount(input: string): number {
+  if (input.length == 0) return 0;
+
+  const cleanedInput = cleanString(input);
+
+  return cleanedInput.split(" ").length;
```
Comment (hayd) [inline strings/word_count.ts]: Does `(input.trim().match(/\s+/g) || []).length` also work here?
(Not sure which performs better.)

Comment (dshubhadeep) [reply strings/word_count.ts]: @hayd Yeah it works, but your code is returning the (no. of words - 1). So, incrementing by 1 will result in the correct output. 
I compared both the methods using [Performance API](https://developer.mozilla.org/en-US/docs/Web/API/Performance/now). There is a negligible difference in the performance

**Comment by ry** [comment]
Thank you for the contribution but this functionality is too specific for deno_std. The criteria I'm using is whether similar functionality is found in Go's standard library. 

Rather than having a `wordCount`, I would accept `strings.fields()` which should operate like this:
https://golang.org/pkg/strings/#Fields

The tests and documentation should be ported from Go: https://github.com/golang/go/blob/b41eee4c8a2fe692c1d9fb46972b9047b5dc02b7/src/strings/strings_test.go#L462-L493

**Comment by dshubhadeep** [comment]
@ry No problem. I will start working on `fields` implementation.

**Comment by ry** [comment]
@dshubhadeep actually I forgot we have [String.split](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/split) already... so I guess fields isn't necessary either.

**Comment by j-f1** [comment]
They behave differently, though.

```js
// input:
const s = "  foo bar  baz   "
// Go:
import "strings"
strings.Fields(s) => ["foo" "bar" "baz"]

// JS
s.split()      => ["  foo bar  baz   "]
s.split("")    => [" ", " ", "f", "o", "o", " ", ..., "z", " ", " ", " "]
s.split(" ")   => ["", "", "foo", "bar", "", "baz", "", "", ""]
s.split(/\s+/) => ["", "foo", "bar", "baz", ""]
```

**Comment by dshubhadeep** [comment]
@ry Oh. Is there anything that needs to be added to the strings module ?

**Comment by ry** [comment]
@j-f1 @dshubhadeep Do you think it's worth it to add "fields" as an alias for `s.split(/\s+/).filter(x => x.length > 0)` ?

**Comment by dshubhadeep** [comment]
@ry No, it just feels like an unnecessary abstraction.

**Pull request closed without merging**